boa/src/builtins/object/mod.rs

```diff
@@ -406,6 +408,28 @@ pub fn make_object(_: &mut Value, args: &[Value], ctx: &mut Interpreter) -> Resu
     Ok(object)
 }
 
+/// Uses the SameValue algorithm to check equality of objects
+pub fn is(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+    let x = args.get(0).expect("Cannot get object").clone();
+    let y = args.get(1).expect("Cannot get object").clone();
+
+    if x.get_type() != y.get_type() {
+        return Ok(Value::boolean(false));
+    }
+
+    let result = match (x.data(), y.data()) {
+        (ValueData::BigInt(x), ValueData::BigInt(y)) => BigInt::same_value(x, y),
+        (ValueData::Rational(x), ValueData::Rational(y)) => Number::same_value(*x, *y),
+        (ValueData::Rational(x), ValueData::Integer(y)) => Number::same_value(*x, f64::from(*y)),
+        (ValueData::Integer(x), ValueData::Rational(y)) => Number::same_value(f64::from(*x), *y),
+        (ValueData::Integer(x), ValueData::Integer(y)) => x == y,
+
+        (_, _) => same_value(&x, &y, false),
+    };
+
+    Ok(Value::boolean(result))
```

Comment:
we can use the `same_value` function here we don't have to reimplement it.

I don't know why `same_value` has a `strict` parameter it is not dependent on strict mode